tests/testthat/test-bettr.R

```diff
@@ -1,56 +1,189 @@
+                 regexp = 'Specified flips must be logical scalars',
+                 fixed = TRUE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(flip = c(TRUE, FALSE)))),
+                 regexp = 'Specified flips must be logical scalars',
+                 fixed = TRUE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(transform = 1))),
+                 regexp = 'Specified transforms must be character scalars',
+                 fixed = FALSE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(transform = "x"))),
+                 regexp = 'Specified transforms must be character scalars',
+                 fixed = FALSE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(transform = c("Rank", "None")))),
+                 regexp = 'Specified transforms must be character scalars',
+                 fixed = FALSE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(cuts = "x"))),
+                 regexp = 'Specified cuts must be numeric vectors',
+                 fixed = TRUE)
+    expect_error(bettr(df = df, 
+                       initialTransforms = list(m1 = list(cuts = FALSE))),
+                 regexp = 'Specified cuts must be numeric vectors',
+                 fixed = TRUE)
+    
+    ## bstheme
+    expect_error(bettr(df = df, bstheme = "missing"),
+                 regexp = "\\'arg\\' should be one of",
+                 fixed = FALSE)
+})
+
+test_that("bettr runs with valid inputs", {
+    set.seed(123)
+    df <- data.frame(Method = rep(c("A", "B", "C"), each = 3),
+                     Metric = rep(c("m1", "m2", "m3"), 3),
+                     Value = runif(n = 9, min = 0, max = 3)) %>%
+        tidyr::spread(key = Metric, value = Value)
+    df$m3 <- paste0(df$m3, "x")
+    metrics <- setdiff(colnames(df), "Method")
+    metricInfo <- data.frame(Metric = c("m1", "m2", "m3"), num = 1:3)
+    idInfo <- data.frame(Method = c("A", "B", "C"), lets = letters[1:3])
+    
+    ## Default parameters
+    app <- bettr(df, idCol = "Method")
+    expect_s3_class(app, "shiny.appobj")
+
+    ## Specify subset of metrics
+    app <- bettr(df, idCol = "Method", metrics = c("m1", "m3"))
+    expect_s3_class(app, "shiny.appobj")
+    
+    ## Specify initial weights
+    app <- bettr(df, idCol = "Method", 
+                 initialWeights = c(m1 = 0.1, m2 = 0.5, m3 = 0.7))
+    expect_s3_class(app, "shiny.appobj")
+    
+    ## Specify initial transforms
+    app <- bettr(df, idCol = "Method", metrics = c("m1", "m3"),
+                 initialTransforms = list(m1 = list(offset = 2, 
+                                                    cuts = 1.5),
+                                          m2 = list(flip = TRUE, 
+                                                    transform = "Rank")))
```

Comment:
I am fine in having the tests at this level - shall we try still (separate branch?) to give shinyTest (or however that framework is called) a shot?
As far as this does not preclude a seamless CRAN/Bioc submission, though

Reply:
Sure - I don't have any experience with that yet, but it's something we can work on on the side I think (doesn't have to affect the other development). I was also thinking that we may want to make sure things are (more) stable before embarking on making more complex tests.

Reply:
Agree, also for me it is a later-thing. A nice addon, but there are other priorities, and the good ol' classical tests on the function in the app do have the front line